pulser-pasqal/pulser_pasqal/job_parameters.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Parameters to build a sequence sent to the cloud."""
+from __future__ import annotations
+
+import dataclasses
+from typing import Mapping, Union
+
+from numpy.typing import ArrayLike
+
+from pulser.register import QubitId
+
+
+@dataclasses.dataclass
+class JobParameters:
+    """Parameters representing a job to build the sequence."""
+
+    parameters: dict[str, Union[ArrayLike, Mapping[QubitId, int]]]
```

Comment:
I think this one shouldn't be a dataclass to allow for variable kwargs. I think it could mimic the signature of `Sequence.build()`, something like
```suggestion
class JobParameters:
    """Parameters representing a job to build the sequence."""

     def __init__(
        self,
        runs: int, 
        qubits: Optional[Mapping[QubitId, int]] = None,
        **vars: Union[ArrayLike, float, int],
        ):
        ...
```

Reply:
I like the idea, the only thing that annoys me is to put `runs` and the other arguments together in the same kwargs. If a variable is named `runs`, the user won't be able to set it.

Reply:
Well, that would still be a problem with your solution, wouldn't it? I think we just need to add `runs` to the list of protected variable names [here](https://github.com/pasqal-io/Pulser/blob/354d07a37d4652ed9e636b39e43c247882987845/pulser-core/pulser/sequence/sequence.py#L513).